flume-ng-core/src/main/java/org/apache/flume/serialization/FlumeEventAvroEventDeserializer.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flume.serialization;
+
+import org.apache.avro.Schema;
+import org.apache.avro.util.Utf8;
+import org.apache.flume.Context;
+import org.apache.flume.Event;
+import org.apache.flume.FlumeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * A deserializer that parses Avro container files, generating one Flume event
+ * per record in the Avro file, and storing binary avro-encoded records in
+ * the Flume event body.
+ */
+public class FlumeEventAvroEventDeserializer extends AbstractAvroEventDeserializer {
+
+  private static final Logger logger =
+          LoggerFactory.getLogger(FlumeEventAvroEventDeserializer.class);
+
+  private FlumeEventAvroEventDeserializer(Context context, ResettableInputStream ris) {
+    this.ris = ris;
+
+    schemaType = AvroSchemaType.valueOf(
```

Comment:
This (line 48-57) is a copy-paste from `AvroEventDeserializer`. Could you please simplify it by pushing it (from here and from the `AvroEventDeserializer`) up to the `AbstractAvroEventDeserializer`? 
If the logic is moved to the parent class then its `schemaType` field can be changed to `private final`.